app/Mage.php

```diff
@@ -47,8 +47,13 @@
 
 Varien_Autoload::register();
 
-include_once "phpseclib/bootstrap.php";
-include_once "mcryptcompat/mcrypt.php";
+/** AUTOLOADER PATCH **/
+if (file_exists($autoloaderPath = BP . DS . 'vendor/autoload.php') ||
+    file_exists($autoloaderPath = BP . DS . '../vendor/autoload.php')
+) {
+    require $autoloaderPath;
+}
+/** AUTOLOADER PATCH **/
```

Comment:
Is thi required for this PR?

Reply:
I added it since now composer is required so also the patch so it seems that we can directly have the autoload built in in our core instead of having composer to constantly patch the app/Mage.php

Reply:
Double check this because I think it might break someone who has set a custom vendor-dir:

https://getcomposer.org/doc/06-config.md#vendor-dir

I think @Flyingmana mentioned this in another PR or issue but I’m on mobile now and can’t find it.

Reply:
@justinbeaty thanks. Never used a custom vendor dir ...

Tested with ...
```
{
    "config": {
        "vendor-dir": "test"
    }
}
```

Patched Mage.php looks like ...

```
/** AUTOLOADER PATCH **/
if (file_exists($autoloaderPath = BP . DS . 'test/autoload.php') ||
    file_exists($autoloaderPath = BP . DS . '../test/autoload.php')
) {
    require $autoloaderPath;
}
/** AUTOLOADER PATCH **/
```

:+1:

Reply:
ah! didn't even know this possibility existed. I'll remove the patch then.